test/unit/deferred.js

```diff
@@ -861,15 +882,24 @@ QUnit.test( "jQuery.when(nonThenable) - like Promise.resolve", function( assert
 QUnit.test( "jQuery.when(thenable) - like Promise.resolve", function( assert ) {
 	"use strict";
 
-	var CASES = 16,
-		slice = [].slice,
+	var customToStringThen = {
+		then: function( onFulfilled ) {
+			onFulfilled();
+		}
+	};
+	if ( typeof Symbol === "function" ) {
```

Comment:
I don't think we care about those cases because they coincide with impossibility of setting @@toStringTag, but I'll see what I can do anyway.